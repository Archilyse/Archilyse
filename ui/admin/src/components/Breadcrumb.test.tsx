import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Breadcrumb } from '.';

afterEach(cleanup);

const MOCK_HIERARCHY = [
  { text: 'All Clients', href: '/some-url-1' },
  { text: 'Client name', href: '/some-url-2' },
  { text: 'Site: some words here', href: '/some-url-3' },
  { text: 'Building: 23241234', href: '/some-url-4' },
  { text: 'Folder - 1', href: '/some-url-5' },
];

it('renders correctly empty breadcrumb', () => {
  const { container } = render(
    <MemoryRouter>
      <Breadcrumb />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

it('renders correctly breadcrumb with hierarchy', () => {
  const { container } = render(
    <MemoryRouter>
      <Breadcrumb hierarchy={MOCK_HIERARCHY} />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
