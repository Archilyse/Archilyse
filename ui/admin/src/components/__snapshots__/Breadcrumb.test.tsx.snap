// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly breadcrumb with hierarchy 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="breadcrumb-link"
          href="/some-url-1"
        >
          <span
            style="max-width: 0;"
          >
            <div
              class="truncate-overflow"
              style="max-width: 0;"
            >
              All Clients
            </div>
            <span
              class="truncate-hidden"
            >
              All Clients
            </span>
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
          style="font-size: 20px; color: rgb(128, 128, 128); margin-left: 0px;"
        >
          navigate_next_rounded
        </span>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="breadcrumb-link"
          href="/some-url-2"
        >
          <span
            style="max-width: 0;"
          >
            <div
              class="truncate-overflow"
              style="max-width: 0;"
            >
              Client name
            </div>
            <span
              class="truncate-hidden"
            >
              Client name
            </span>
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
          style="font-size: 20px; color: rgb(128, 128, 128); margin-left: 0px;"
        >
          navigate_next_rounded
        </span>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="breadcrumb-link"
          href="/some-url-3"
        >
          <span
            style="max-width: 0;"
          >
            <div
              class="truncate-overflow"
              style="max-width: 0;"
            >
              Site: some words here
            </div>
            <span
              class="truncate-hidden"
            >
              Site: some words here
            </span>
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
          style="font-size: 20px; color: rgb(128, 128, 128); margin-left: 0px;"
        >
          navigate_next_rounded
        </span>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="breadcrumb-link"
          href="/some-url-4"
        >
          <span
            style="max-width: 0;"
          >
            <div
              class="truncate-overflow"
              style="max-width: 0;"
            >
              Building: 23241234
            </div>
            <span
              class="truncate-hidden"
            >
              Building: 23241234
            </span>
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
          style="font-size: 20px; color: rgb(128, 128, 128); margin-left: 0px;"
        >
          navigate_next_rounded
        </span>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="breadcrumb-link"
        >
          <span
            style="max-width: 0;"
          >
            <div
              class="truncate-overflow"
              style="max-width: 0;"
            >
              Folder - 1
            </div>
            <span
              class="truncate-hidden"
            >
              Folder - 1
            </span>
          </span>
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
            style="font-size: 20px; color: rgb(128, 128, 128); margin-left: 4px;"
          >
            keyboard_arrow_down_rounded
          </span>
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`renders correctly empty breadcrumb 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    />
  </nav>
</div>
`;
